### ISSUE TITLE
Pilot LFX Insights On The Express Project

### ISSUE THREAD

**Issue by bensternthal** [expressjs/discussions#205]
I was chatting with Ulises earlier this week about some of the metrics we have available to us in LFX (the Linux Foundation community tool) and we both thought the Express project would be a good candidate to test out LFX Insights. If this works well, we'd like to roll it out to other OpenJS projects. 

**What Is LFX Insights**
Insights provides projects with more visibility into their community dynamics, particularly around contributors, organizations, issues, and pull request metrics. It does this through an app (similar to the DCO bot) installed at the organization level. This is then enabled for one or more repos under the organization. Insights uses [crowd.dev](http://crowd.dev/) under the hood to gather read-only data about specified public repositories in a GitHub organization.

**What Information Is Collected (only public information from Github)**

- Stars/Un-stars
- Forks
- Issues
- Pull requests
- Discussions
- Comments on issues/pull requests/discussions
- Closing of issues/pull requests/discussions

**Future Ideas**
We’d like to use insights to display security metrics for OpenJS projects including those being currently developed as part of the Sovereign Tech Fund initiative/score cards/ best practices etc. Use this data to help inform which if any projects should be Emeritus. 

**What I Need From The Project**
I’d like a) A thumbs up from the TSC if we are OK to proceed b) Any question or concerns C) Who from Express should have access to the data (community managers)

**How Do We Turn This On**
I need to be made an admin (this can be temporary) on the Express org, this will allow me to install the read only Linux Foundation app. Once I install the app my privileges can be revoked.

**More Info**
You can find [Insights documentation here](https://docs.linuxfoundation.org/lfx/insights/v3-beta-version-current).

@UlisesGascon @wesleytodd

**Comment by UlisesGascon** [comment]
Initial Big +1 from my side. I will check the documentation

**Comment by sheplu** [comment]
+1 on my side, it would be great to have a global overview and if this also provide the changes (weekly? monthly?) this would be even better

**Comment by wesleytodd** [comment]
Those orgs are a part of the Express project and governance, so they are the same as this org. Now that you are asking that though I wonder if all the docs line up with that lol. It is not helpful to us if we cannot have all of them IMO. It is not too hard to do this kind of thing across one org, it is the fact we are in many that is our main issue I think.

**Comment by bensternthal** [comment]
@wesleytodd heh I don't think the docs in those other projects align :) However it is no problem to add these into our system as projects under express, we can then do a rollup of the metrics.

**Comment by bensternthal** [comment]
Ok, folks, I am going to be adding the additional orgs into LFX today (no action needed from anyone for that step).

Once that is done, I will need admin access so that I can enable the LF app on each org. 

Once the project approves can you grant me this access?

**Comment by wesleytodd** [comment]
Invites sent.

**Comment by bensternthal** [comment]
@wesleytodd I got the invites for pillarjs, & jshttp, but not for express. If you can add me to that last one I can get this setup for y'all. Much thanks!!

**Comment by wesleytodd** [comment]
Oh sorry! I thought we had already added you on that. Will do it now.

**Comment by bensternthal** [comment]
I am in! OK I will report back when this is setup.

**Comment by bensternthal** [comment]
Folks, we've setup metrics for the three projects and I wanted to share what the dashboard looks like. Note I have a ticket filed to get the rollup working correctly, I will update folks when that is resolved:

https://insights.lfx.linuxfoundation.org/foundation/ojsf/overview?project=ojsf-express&

Please poke around and reach out either here or via slack if you have questions!

**Comment by UlisesGascon** [comment]
I have a question regarding grouping or identification. I can see that sometimes the same contributors are shown twice like Wes or myself. Is there something that we can do to unify our identities?

![Screenshot from 2024-04-05 22-13-26](https://github.com/expressjs/discussions/assets/5110813/13c258a3-54b8-44d6-b955-5f9790fcd3e4)

**Comment by bensternthal** [comment]
Intriguing, let me poke around on this one.

**Comment by inigomarquinez** [comment]
I love this initiative! Thanks @bensternthal !

**Comment by wesleytodd** [comment]
I am pretty sure I have not been correctly switching from my netflix git identity to my personal one for much of this work lately. I have both emails on my account so GH does it correctly but I likely need to start remembering to swap my profiles. Ideally I could get a computer just to work on OSS so I am not adding even more annoyance/time sinks to my workflow, but that is not likely to happen any time soon.

**Comment by ljharb** [comment]
I merged them so i don’t have to switch - you have to file an IT ticket for that tho.

**Comment by UlisesGascon** [comment]
Side note... I was exploring how to collect more detailed data about the GitHub activity for Node (see: https://github.com/nodejs/next-10/issues/271#issuecomment-2164871105) and I also took the opportunity to build a combined report for Express/Pillarjs/jshttp ([see](https://cauldron.io/project/8456?from_date=2023-06-13&to_date=2024-06-13&tab=overview)).

The insights that this tool provides are different and complementary to the LFX. For Example you can get a view of the most relevant CHAOSS metrics for us. You can easily see the impact since #160

**Comment by bjohansebas** [comment]
I'm going to close this because the integration is already done, and I don't see any actionable items to address. But feel free to reopen it if there's anything to be done.
